annotation-processor/src/test/java/org/hibernate/validator/ap/ConstraintValidationProcessorTest.java

```diff
@@ -96,6 +98,35 @@ public void fieldLevelValidationUsingBuiltInConstraints() {
 		);
 	}
 
+
+	@Test
+	@TestForIssue( jiraKey = "HV-840" )
+	public void overridingMethodParameterConstrainsTest() {
+		File sourceFile = compilerHelper.getSourceFile( MethodOverridingTests.class );
+
+		boolean compilationResult = compilerHelper.compile( new ConstraintValidationProcessor(), diagnostics, sourceFile );
+
+		assertFalse( compilationResult );
+		assertThatDiagnosticsMatch(
+				diagnostics,
+				new DiagnosticExpectation( Kind.ERROR, 29 ),
+				new DiagnosticExpectation( Kind.ERROR, 47 ),
+				new DiagnosticExpectation( Kind.ERROR, 63 ),
+				new DiagnosticExpectation( Kind.ERROR, 82 ),
+				// commented out as method return values annotations are not processed yet
+				// new DiagnosticExpectation( Kind.ERROR, 117 ),
```

Comment:
In the original task only constraints on parameters  were mentioned, but in the documentation which you pointed out in your first comment I saw also constraints on method return values. So I've added the test case for it. 

I suppose we can add this functionality in this PR. Because to implement it we will need a lot of functionality present in current `MethodParameterOverrideCheck` (all what is related to finding overridden methods). So I can see two way we can go to implement it:

- first one will be to add a methodCheck implementation to `MethodParameterOverrideCheck`, which will validate annotations on return values of overridden methods. And probably rename `MethodParameterOverrideCheck` to something more suitable for validating both parameters and return values.
- second one - move all common functionality (related to finding overridden methods) to some abstract class and provide to sub classes one for method parameter checks and another for method return value checks.

I didn't want to add it all until we make sure that what is already implemented is correct :)